src/qat/model/device.py

```diff
@@ -1,12 +1,75 @@
 from __future__ import annotations
 
+import uuid
 from typing import Optional
 
-from pydantic import Field
+import numpy as np
+from pydantic import ConfigDict, Field, model_validator
 
-from qat.model.component import Component
-from qat.model.serialisation import Ref, RefDict, RefList
-from qat.purr.compiler.devices import ChannelType
+from qat.utils.pydantic import WarnOnExtraFieldsModel
+
+QubitId = int
+
+
+class Component(WarnOnExtraFieldsModel):
+    """
+    Base class for any logical object which can act as a target of a quantum action
+    - a Qubit or various channels for a simple example.
+
+    Attributes:
+        uuid: The unique string representation of the component.
+    """
+
+    model_config = ConfigDict(validate_assignment=True, extra="ignore")
+
+    uuid: str = Field(default_factory=lambda: str(uuid.uuid4()), frozen=True)
```

Comment:
Still not sure if we should keep the `uuid` or not. This question will probably answer itself when we merge the functionality of these hardware models with the new pipeline stuff.

Reply:
I think you're right, keep it until we know we don't need it (although because of the hashing, I expect that this will still be useful!)